### PR TITLE
refactor(writers): Don't leak timers

### PR DESCRIPTION
**Pull request by disq**
Alternative to https://github.com/cloudquery/plugin-sdk/pull/1058

### CODE REVIEW

**Comment by hermanschaaf** [comment]
I haven't looked into it yet, but what does `time.Ticker` do if you pass in a 0 duration? If that works, maybe we can simplify by using that?

**Comment by disq** [comment]
> I haven't looked into it yet, but what does `time.Ticker` do if you pass in a 0 duration? If that works, maybe we can simplify by using that?

Looks like it will fire immediately (after a certain unspecified) delay:

```go
func when(d Duration) int64 {
	if d <= 0 {
		return runtimeNano()
	}
```

**Comment by disq** [comment]
We could also use `.Reset(time.Duration)` to re-use the same timer over and over again, would require a follow-up PR though... cc @candiduslynx

File: writers/batchwriter/batchwriter.go [hunk 1]
```diff
@@ -145,7 +148,8 @@ func (w *BatchWriter) worker(ctx context.Context, tableName string, ch <-chan *m
 				w.flushTable(ctx, tableName, resources)
 				resources, sizeBytes = resources[:0], 0
 			}
-			tick = timer(w.batchTimeout)
+			tickClose()
```
Comment (candiduslynx) [inline writers/batchwriter/batchwriter.go:151]: you shouldn't need to manually close on every iteration

**Review by candiduslynx — CHANGES_REQUESTED** [review]

File: writers/batchwriter/batchwriter.go (hunk 1 above)
Comment (disq) [reply writers/batchwriter/batchwriter.go:151]: incoming commit:

```go
type ticker interface {
	Chan() <-chan time.Time
	Reset(time.Duration)
	Stop() bool
}
```

**Comment by candiduslynx** [comment]
@disq WDYT about https://github.com/cloudquery/plugin-sdk/pull/1058/commits/e63bc5e260085294691b371f717a164267b451cf instead?

**Pull request closed without merging**